scripts/Run-Self-Hosted-Runner-Test.ps1

```diff
@@ -0,0 +1,261 @@
+{
+    Param(
+        # Drive Specification in the form of a single alphabet string. (eg. "C:")
+        [Parameter(Mandatory = $True)] [string] $DriveSpecification
+    )
+
+    if (($DriveSpecification.Length -eq $Null) -or ($DriveSpecification.Length -ne 2)) {
+        ThrowWithErrorMessage -ErrorMessage "*** ERROR *** No drive or Invalid drive specified."
+    }
+
+    # Convert drive to single letter (eg. "C:" to "C") for Get-Volume.
+    $DriveSpecification = $DriveSpecification -replace ".$"
+    $FreeSpaceGB = (((Get-Volume $DriveSpecification).SizeRemaining) / 1GB).ToString("F2")
+
+    return $FreeSpaceGB
+}
+
+if ($VerbosePreference -eq 'Continue') {
+    Write-Log "Command               : $TestCommand"
+    Write-Log "Arguments             : $TestArguments"
+    Write-Log "Test Hang Timeout     : $TestHangTimeout"
+    Write-Log "User mode dump Folder : $UserModeDumpFolder"
+    Write-Log "Kernel dump needed    : $NeedKernelDump"
+}
+
+# Verify timeout is non-zero.
+if ($TestHangTimeout -le 0) {
+    ThrowWithErrorMessage `
+    -ErrorMessage "*** ERROR *** Invalid test hang timeout value: $TestHangTimeout"
+}
+
+# Verify user mode dump folder name is not null.
+if ($UserModeDumpFolder -eq $Null) {
+    ThrowWithErrorMessage `
+    -ErrorMessage "*** ERROR *** User mode dump folder cannot be NULL"
+}
+
+# Create dump folder if not present.
+if (-not (Test-Path -Path $UserModeDumpFolder)) {
+    New-Item -Path $UserModeDumpFolder -ItemType Directory -Force | Out-Null
+
+    # Verify dump folder creation.
+    if (-not (Test-Path -Path $UserModeDumpFolder)) {
+        ThrowWithErrorMessage `
+        -ErrorMessage "*** ERROR *** User mode dump folder creation failed: $UserModeDumpFolder"
+    }
+}
+
+# Check if procdump64.exe and notmyfault64.exe are present on the system.
+$ProcDumpBinary = "ProcDump64.exe"
+Write-Log "Verifying $ProcDumpBinary presence in $Pwd..."
+$ProcDumpBinaryPath = GetToolLocationPath -ToolName $ProcDumpBinary
+Write-Log "$ProcDumpBinary location: $ProcDumpBinaryPath"
+Write-Log "`n"
+
+$NotMyFaultBinary = "NotMyFault64.exe"
+Write-Log "Verifying $NotMyFaultBinary presence in $Pwd..."
+$NotMyFaultBinaryPath = GetToolLocationPath -ToolName $NotMyFaultBinary
+Write-Log "$NotMyFaultBinary location: $NotMyFaultBinaryPath"
+Write-Log "`n"
+
+# While at it, enable EULA for all SysInternals tools.
+REG ADD HKCU\Software\Sysinternals /v EulaAccepted /t REG_DWORD /d 1 /f | Out-Null
+
+# The following 'Set-ItemProperty' command enables a full memory dump.
+# Not enabled yet as it needs a VM with an explicitly created page file of at least (physical_memory + 1MB) in size.
+# The default value of the 'CrashDumpEnabled' key is 7 ('automatic' sizing of dump file size (system determined)).
+# https://learn.microsoft.com/en-us/troubleshoot/windows-server/performance/memory-dump-file-options
+#
+# Set-ItemProperty -Path 'HKLM:\SYSTEM\CurrentControlSet\Control\CrashControl' -Name 'CrashDumpEnabled' -Value 1
```

Comment:
A full memory dump might not al;ways be needed, we could later add some conditional logic to select it

Reply:
I disagree.  In my experience it is always better to have the full dump, especially in post-mortem scenarios where the root cause is not immediately obvious and having all possible kernel info comes in handy.